File: webclient/src/app/models/pilot.ts
import { Date } from './date';
import { Flight } from './flight';
import { FlightCompany } from './flightcompany';

export class Pilot
{
    private id:number;
    private name: string;
    private surname: string;
    private phone: number;
    private birthday:String;
    private gender: string;
    private weight: number;
    private height: number;
    private flightHistory: Flight[];
    private rank:string;
    private flightCompany:FlightCompany;
    private workExperience: number;
    private salary: number;
    private nextFlights: Flight[];

    public constructor(id:number,name:string,surname:string,phone:number,birthday:String,gender:string,weight:number,
                        height:number,rank:string,flightCompany:FlightCompany,workExperience:number,salary:number)
    {
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.phone = phone;
        this.birthday = birthday;
        this.gender = gender;
        this.weight = weight;
        this.height = height;
        this.rank = rank;
        this.flightCompany = flightCompany;
        this.workExperience = workExperience;
        this. salary = salary;
    }

    public addFligtToFlightHistory(flight:Flight)
    {
        this.flightHistory.push(flight);
    }

    public addFlightToNextFlight(flight:Flight)
    {
        this.nextFlights.push(flight);
    }

    public get getiId(): number {
        return this.id;
    }

    public set setId(value: number) {
        this.id = value;
    }

    public get getName(): string {
        return this.name;
    }

    public set setName(value: string) {
        this.name = value;
    }

    public get getSurname(): string {
        return this.surname;
    }

    public set setSurname(value: string) {
        this.surname = value;
    }
    public get getPhone(): number {
        return this.phone;
    }
    
    public set setPhone(value: number) {
        this.phone = value;
    }

    public get getBirthday(): String {
        return this.birthday;
    }

    public set setBirthday(value: String) {
        this.birthday = value;
    }

    public get getGender(): string {
        return this.gender;
    }

    public set setGender(value: string) {
        this.gender = value;
    }

    public get getWeight(): number {
        return this.weight;
    }

    public set setWeight(value: number) {
        this.weight = value;
    }

    public get getHeight(): number {
        return this.height;
    }

    public set setHeight(value: number) {
        this.height = value;
    }

    public get getFlightHistory(): Flight[] {
        return this.flightHistory;
    }

    public get getRank(): string {
        return this.rank;
    }

    public set setRank(value: string) {
        this.rank = value;
    }

    public get getFlightCompany(): FlightCompany {
        return this.flightCompany;
    }

    public set setFlightCompany(value: FlightCompany) {
        this.flightCompany = value;
    }

    public get getWorkExperience(): number {
        return this.workExperience;
    }

    public set setWorkExperience(value: number) {
        this.workExperience = value;
    }

    public get getSalary(): number {
        return this.salary;
    }

    public set setSalary(value: number) {
        this.salary = value;
    }

    public get getNextFlights(): Flight[] {
        return this.nextFlights;
    }

}

File: src/Pilot.java
import java.util.LinkedList;


public class Pilot extends Person implements Employee{
    private String rank;
    private FlightCompany flightCompany;
    private int workExperience;
    private int salary;
    private Queue nextFlights;

    public Pilot(String name, String surname, int idNumber, Phone phone, Date birthday, String gender, double weight, double height, String rank, FlightCompany flightCompany, int workExperience, int salary) {
        super(name, surname, idNumber, phone, birthday, gender, weight, height);
        this.rank = rank;
        this.flightCompany = flightCompany;
        this.workExperience = workExperience;
        this.salary = salary;
        this.nextFlights = new Queue();
    }

    @Override
    public void addFlightToNextFlight(Flight flight) {
        this.nextFlights.enqueue(flight);
    }

    @Override
    public boolean isAvailable(Flight flight) {
        return false;
    }

    @Override
    public void addFligt(Flight flight) {
        addFlightToFlightHistory(flight);
    }


    public String getRank() {
        return rank;
    }
    public FlightCompany getFlightCompany() {
        return this.flightCompany;
    }


    public int getWorkExperience() {
        return workExperience;
    }

    public void setWorkExperience(int workExperience) {
        this.workExperience = workExperience;
    }

    public int getSalary() {
        return salary;
    }

    public void setSalary(int salary) {
        this.salary = salary;
    }

    public Queue getNextFlights() {
        return nextFlights;
    }

    public void setRank(String rank) {
        this.rank = rank;
    }


}